Source/WebCore/rendering/svg/LegacyRenderSVGForeignObject.cpp

```diff
@@ -54,8 +56,22 @@ SVGForeignObjectElement& LegacyRenderSVGForeignObject::foreignObjectElement() co
     return downcast<SVGForeignObjectElement>(RenderSVGBlock::graphicsElement());
 }
 
+bool LegacyRenderSVGForeignObject::isUnderSVGRoot() const
+{
+    auto parent = this->parent();
+    while (parent) {
+        if (is<RenderSVGRoot>(parent) || is<LegacyRenderSVGRoot>(parent))
+            return true;
+        parent = parent->parent();
+    }
+    return false;
```

Comment:
```suggestion
    return ancestorsOfType<LegacyRenderSVGRoot>(*this).first();
```

I don't think it's possible to have RenderSVGRoot and LegacyRenderSVGForeignObject in the same tree. Either everything is legacy (LBSE disabled) or non-legacy (LBSE enabled).

Checking for LegacyRenderSVGRoot should be sufficient.

Also you can use ancestorsOfType instead of this loop.

Reply:
Although, I'm surprised there isn't a pre-existing function to get the SVG root, seems like something that would be done fairly often. Maybe Said or Nikolas know if there is one?

Reply:
There is the legacy SVGRenderSupport::findTeeeRootObject() method. However I’d use ancestorsOfType for new code…

Reply:
I think we should use `SVGRenderSupport::findTreeRootObject()` in that case, I would consider `LegacyRenderSVGForeignObject` to be old code.  This fix does not affect LBSE.